### DISCUSSION TITLE
gRPC go-kit stream

### DISCUSSION THREAD

**Discussion by maeglindeveloper** [go-kit/kit#1213] [Q&A]
Hey everyone, hope this message finds you well :+1: 

I've seen multiple issues talking about how to handle `gRPC` stream in go-kit, and it was a long time ago.
I'm wondering if there is anything new on this, or if there is somewhere some example of how to implement it.

Ref issues: https://github.com/go-kit/kit/issues/935, https://github.com/go-kit/kit/issues/899, https://github.com/go-kit/kit/issues/624

I did try to implement it but I'm confused when I have to specify the `encode/decode` func for both server & client :-1: since we are using a `stream`.

Well if anyone have some experience on this, I'm fully opened to it :+1: 

cc @peterbourgon ;) !

**Comment by maeglindeveloper** [comment]
Hey @peterbourgon , any advice on this [🙂](https://emojipedia.org/slightly-smiling-face/) ?

**Comment by peterbourgon** [comment]
Go kit explicitly targets RPC semantics. gRPC streaming isn't a match for Go kit endpoints.
